tests/tests/test_demo_artifact.py

```diff
@@ -79,31 +89,62 @@ def run_demo_script_up():
     def test_demo_artifact(self, run_demo_script):
         """Tests that the demo script does indeed upload the demo Artifact to the server."""
 
+        # Give the tests a clean docker-compose environment
         stop_docker_compose()
 
         logging.info("--------------------------------------------------")
         logging.info("Running test_demo_artifact_upload")
         logging.info("--------------------------------------------------")
-        self.demo_artifact_upload(run_demo_script)
-        stop_docker_compose()
+        demo_script = run_demo_script()
+        try:
+            self.demo_artifact_upload(demo_script)
+        except Exception as e:
```

Comment:
mmm... you want to except only `DemoScriptException`, right?

Reply:
Discussed in person.

Better to then use a try/finally. Better to double check in Google